src/server/biome.ts

```diff
@@ -3,20 +3,26 @@ import { getAdjacentCoords } from './utils';
 
 export class TileType {
   public type: string;
+  public vegetation: [number, string | null];
   public isMountain: boolean;
   public isWater: boolean;
   public isOcean: boolean;
   public isRiverGen: boolean;
   public heightClass: number; // 0 = Sea Level, 1 = Lowlands, 2 = Plains, 3 = Highlands, 4 = Mountains, 5 = Mountain Peak
 
-  constructor(type: string, heightClass: number, isWater = false, isOcean = false, isRiverGen = false) {
+  constructor(type: string, heightClass: number, vegetation: [number, string] | null = null, isWater = false, isOcean = false, isRiverGen = false) {
     this.type = type;
+    this.vegetation = vegetation || [0, null];
```

Comment:
Isn't it more semantically correct to write `vegetation ?? [0, null]`?

Reply:
I believe that wouldn't work since vegetation is an array...